### DISCUSSION TITLE
Rendering plantuml / mermaid in markdowns included with !docs

### DISCUSSION THREAD

**Discussion by gparmeggiani** [structurizr/lite#82] [Q&A]
### Question

I'm trying to render a plantuml diagram in a markdown file included with `!docs`, however it doesn't render and I simply see the contents of the plantuml script.

Is this a supported feature?

Here's the example I'm trying to render:

```plantuml
@startuml 

Alice -> Bob: Hello!
Bob -> Alice: Hi!
		
@enduml
```

**Comment by simonbrowndotje** [comment]
You have a couple of options:

1. [Image views](https://structurizr.com/help/image-views).
2. You can write a DSL plugin to pre-process the PlantUML in your Markdown files ... see https://github.com/structurizr/examples/tree/main/dsl/plantuml-and-mermaid for an example.

**Reply by gparmeggiani** [reply]
thanks for the reply.   

The plugin option achieves exactly what I need, as I can zoom-in from a component to the code where I can describe various aspects of the component with text and multiple plantuml diagrams. 

Can the same be achieved with a script instead of a plugin? or could the plantuml / mermaid support be integrated? I'm asking because I'm thinking on how to distribute the plugin without committing to the repo the .jar file

**Comment by DaveSlinn** [comment]
I know this discussion is in the Lite repo, and it's deprecated in favour of the Structurizr vNext, but I'm still a little unclear on how to enable a Mermaid plugin.  I've read the docs, but it *sounds* like I need to create my own plugin that implements StructurizrDslPlugin, but I see code in the primary repo that does this: [Mermaid.java](https://github.com/structurizr/structurizr/blob/main/structurizr-application/src/main/java/com/structurizr/dsl/plugin/documentation/Mermaid.java).  I've added !plugin com.structurizr.dsl.plugin.documentation.Mermaid.jar to my workspace, but I get an error that it can't be found.  Do I need to create my own copy of this jar file, or does it already exist?  I am not a Java developer, so I don't have the experience to be making .jar files.

**Reply by simonbrowndotje** [reply]
The PlantUML and Mermaid plugins are not included in Lite. It's certainly possible to extract them from the vNext repo, compile, and install them into Lite, but I'd recommend just using the new tooling.

**Reply by DaveSlinn** [reply]
I've switched to the vNext docker image, but my mermaid scripts are not rendering - just showing the diagram text.
<img width="972" height="992" alt="image" src="https://github.com/user-attachments/assets/ca873e4d-3697-430e-b39a-7cc253cb897f" />
Do I have to add a plugin reference in my DSL to enable?

**Reply by DaveSlinn** [reply]
I thought I was making headway but am still not getting mermaid to render as an image.  Here is my workspace:
<img width="647" height="942" alt="image" src="https://github.com/user-attachments/assets/dd1657ac-ddd1-4164-b451-4c67b962003f" />
I've confirmed that the local mermaid.ink service is running ok:
<img width="861" height="619" alt="image" src="https://github.com/user-attachments/assets/e5fc7b77-2225-463b-9365-fe3f7824028c" />
But my docs are still rendering like so:
<img width="1444" height="772" alt="image" src="https://github.com/user-attachments/assets/51a9a7a2-401e-49c3-9ab2-71c35a52cb40" />
For this markdown file:
<img width="974" height="511" alt="image" src="https://github.com/user-attachments/assets/b3b20360-1a60-430c-8874-165f0396153e" />
